tests/test_cgi.py

```diff
@@ -58,8 +64,8 @@ def test_checklink (self):
 
     def test_application (self):
         form = dict(url="http://www.example.com/", level="0")
-        formdata = urllib.urlencode(form)
-        environ = {'wsgi.input': BytesIO(formdata)}
+        formdata = urllib_parse.urlencode(form)
+        environ = {'wsgi.input': StringIO(formdata)}
```

Comment:
The WSGI spec ([PEP-3333](https://www.python.org/dev/peps/pep-3333/)) says wsgi.input is supposed to be a stream that returns bytestrings.  StringIO is wrong here, it should be BytesIO.